openquakeplatform/openquakeplatform/static/js/irv_viewer_d3_tree.js

```diff
@@ -159,21 +245,26 @@
                     if (projectDef.children[i].name == tempNewWeight[0]) {
                         projectDefUpdated.children[i].weight = tempNewWeight[1];
                     } else {
-                        for (var j = 0; j < projectDef.children[i].children.length; j++, ct++) {
-                            if (projectDef.children[i].children[j].name == tempNewWeight[0]) {
-                                projectDefUpdated.children[i].children[j].weight = tempNewWeight[1];
-                            } else {
-                                try {
-                                    for (var g = 0; g < projectDef.children[i].children[j].children.length; g++) {
-                                        if (projectDef.children[i].children[j].children[g].name == tempNewWeight[0]) {
-                                            projectDef.children[i].children[j].children[g].weight = tempNewWeight[1];
+                        try {
+                            for (var j = 0; j < projectDef.children[i].children.length; j++, ct++) {
+                                if (projectDef.children[i].children[j].name == tempNewWeight[0]) {
+                                    projectDefUpdated.children[i].children[j].weight = tempNewWeight[1];
+                                } else {
+                                    try {
+                                        for (var g = 0; g < projectDef.children[i].children[j].children.length; g++) {
+                                            if (projectDef.children[i].children[j].children[g].name == tempNewWeight[0]) {
+                                                projectDef.children[i].children[j].children[g].weight = tempNewWeight[1];
+                                            }
                                         }
+                                    } catch (e) {
+                                        // continue
                                     }
-                                } catch (e) {
-                                    // continue
                                 }
                             }
+                        } catch (e) {
+                            // continue
```

Comment:
They are required because there are cases when there is a a child and other cases when a child does not exist